base/compiler/types.jl

```diff
@@ -220,8 +220,6 @@ method_table(ai::AbstractInterpreter) = InternalMethodTable(get_world_counter(ai
 # - inferring non-concrete toplevel call sites
 bail_out_call(interp::AbstractInterpreter, @nospecialize(t), sv)      = t === Any
 bail_out_apply(interp::AbstractInterpreter, @nospecialize(t), sv)     = t === Any
-bail_out_statement(interp::AbstractInterpreter, @nospecialize(t), sv) = t === Bottom
-bail_out_local(interp::AbstractInterpreter, @nospecialize(t), sv)     = t === Bottom
```

Comment:
I added the response to this change [here](https://github.com/JuliaLang/julia/pull/39439#discussion_r574252499).